### ISSUE TITLE
[BUG] "avoid-returning-widgets" does not work correctly with unnamed Widget class parameters

### ISSUE THREAD

**Issue by mit-73** [dart-code-checker/dart-code-metrics#990]
* **Dart code metrics version: 4.17.0**
* **Dart sdk version: ">=2.17.6 <3.0.0"**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  metrics-exclude:
    - test/**
  rules-exclude:
    - test/**
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 20
    maintainability-index: 50
    maximum-nesting: 5
    number-of-parameters: 5
    source-lines-of-code: 50
    technical-debt:
      threshold: 1
      todo-cost: 4
      ignore-cost: 8
      ignore-for-file-cost: 16
      as-dynamic-cost: 16
      deprecated-annotations-cost: 2
      file-nullsafety-migration-cost: 2
      unit-type: "hours"
  rules:
    - avoid-global-state
    - avoid-nested-conditional-expressions:
        acceptable-level: 2
    - avoid-non-null-assertion
    - avoid-unused-parameters
    - avoid-unnecessary-type-assertions
    - avoid-unnecessary-type-casts
    - avoid-global-state
    - binary-expression-operand-order
    - double-literal-format
    - newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    - no-equal-then-else
    - no-magic-number
    - no-object-declaration
    - prefer-conditional-expressions
    - prefer-correct-type-name
    - prefer-first
    - prefer-last
    - always-remove-listener
    - avoid-returning-widgets
    - avoid-unnecessary-setstate
    - avoid-wrapping-in-padding
    - prefer-const-border-radius
    - prefer-extracting-callbacks

```
</details>

**What did you do? Please include the source code example causing the issue.**

```dart
import 'package:flutter/material.dart';

const List<String> items = <String>['a', 'b', 'c'];

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: CustomScrollView(
          slivers: <Widget>[
            SliverGrid(
              gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
                crossAxisCount: 2,
                mainAxisSpacing: 40,
                mainAxisExtent: 297,
                crossAxisSpacing: 8,
              ),
              delegate: SliverChildBuilderDelegate(
                childCount: items.length,
                (BuildContext context, int index) => itemBuilder(items[index]),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget itemBuilder(String v) {
    return Text(v);
  }
}

```

**What did you expect to happen?**
I don't expect to see a warning from a lint

**What actually happened?**
I get a warning from lint on `itemBuilder`: `avoid-returning-widgets`

**Are you willing to submit a pull request to fix this bug?**
Yes, if I can find the time.

**Additional Information**
With a named parameter there is no such problem, as an example `GridView.builder`:

```dart
import 'package:flutter/material.dart';

const List<String> items = <String>['a', 'b', 'c'];

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: GridView.builder(
          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
            crossAxisCount: 2,
            mainAxisSpacing: 40,
            mainAxisExtent: 297,
            crossAxisSpacing: 8,
          ),
          itemCount: items.length,
          itemBuilder: (BuildContext context, int index) => itemBuilder(items[index]),
        ),
      ),
    );
  }

  Widget itemBuilder(String v) {
    return Text(v);
  }
}
```

**Comment by mit-73** [comment]
As I see it, the problem is in [this line of code](https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_returning_widgets/visit_declaration.dart#L29-L33).

**Comment by incendial** [comment]
Available in 4.18.0 🚀
